math/iterative_factorial.cpp

```diff
@@ -5,22 +5,28 @@
  * @author [Renjian-buchai](https://github.com/Renjian-buchai)
  *
  * @description Calculates factorial iteratively.
+ * https://en.wikipedia.org/wiki/Factorial
+ *
+ * Factorial formula:
+ * n! = n * (n - 1) * (n - 2) * ... * 2 * 1
```

Comment:
please use the [one](https://github.com/TheAlgorithms/C-Plus-Plus/pull/2453/commits/4ce13494136d6e60fae4f8c153bf44fd68feb86c#r1181082143) I've commented as this does not render on doxygen
```suggestion
 * \f[n! = n\times(n-1)\times(n-2)\times(n-3)\times\ldots\times3\times2\times1
 *       = n\times(n-1)!\f]
 * for example:
 * \f$4! = 4\times3! = 4\times3\times2\times1 = 24\f$
```